crypto/stack/stack.c

```diff
@@ -305,15 +308,8 @@ void *OPENSSL_sk_delete_ptr(OPENSSL_STACK *st, const void *p)
 
 void *OPENSSL_sk_delete(OPENSSL_STACK *st, int loc)
 {
-    if (st == NULL) {
-        ERR_raise(ERR_LIB_X509, ERR_R_PASSED_NULL_PARAMETER);
+    if (st == NULL || loc < 0 || loc >= st->num)
         return NULL;
-    }
-    if (loc < 0 || loc >= st->num) {
-        ERR_raise_data(ERR_LIB_X509, ERR_R_PASSED_INVALID_ARGUMENT,
-                       "loc=%d", loc);
-        return NULL;
-    }
 
```

Comment:
Note that these changes essentially collapse `OPENSSL_sk_delete()` into `internal_delete()`.
It looks to me that the distinction between the two functions was that `OPENSSL_sk_delete()` provides not only error detection, but also error reporting.

Reply:
I do not think so. The internal delete is without error checking completely and it should stay so.